pkg/agent/agent.go

```diff
@@ -423,19 +432,28 @@ func newHandlers() []handler.Handler {
 	handlers := []handler.Handler{
 		handler.NewAPISvcHandler(agent.cacheManager),
 		handler.NewInstanceHandler(agent.cacheManager),
-		handler.NewCategoryHandler(agent.cacheManager),
 		handler.NewAgentResourceHandler(agent.agentResourceManager),
-		handler.NewCRDHandler(agent.cacheManager),
-		handler.NewARDHandler(agent.cacheManager),
-		// handler.NewACLHandler(agent.cacheManager),
 		agent.proxyResourceHandler,
 	}
 
+	if agent.cfg.GetAgentType() == config.DiscoveryAgent {
+		handlers = append(
+			handlers,
+			handler.NewCategoryHandler(agent.cacheManager),
+			handler.NewCRDHandler(agent.cacheManager),
+			handler.NewARDHandler(agent.cacheManager),
+			// handler.NewACLHandler(agent.cacheManager),
```

Comment:
@jcollins-axway is this still needed?

Reply:
I kept it to enable again when we do fix the update handling of ACLs on API Server